javascript/static-sites-in-vanilla-js.md

```diff
@@ -0,0 +1,167 @@
+important context, particularly while debugging.
+
+Vanilla JavaScript can also be configured to match Gatsby's live reloading with
+[watchify](https://www.npmjs.com/package/watchify).
+
+N.b., [our `new-vanilla-js-app` cookiecutter](../docker/templates/new-vanilla-js-app)
+is set up with a `develop` script that runs a local server and uses `watchify`
+to rebundle and retranspile JavaScript as it changes, as well as a `build`
+script to bundle and transpile code during deployment.
+
+#### Deployment
+
+Vanilla JavaScript can be deployed on GitHub pages, a feature none of our
+other development stacks offers. If a staging site is needed, vanilla JavaScript
+can be deployed on Netlify, the same as our Gatsby apps.
+
+#### Browser compatibility
+
+Gatsby [comes with broad browser support out of the box](
+https://www.gatsbyjs.com/docs/how-to/custom-configuration/browser-support/),
+while vanilla JavaScript apps must add a build step to transpile code with
+Babel.
+
+N.b. (redux), [our `new-vanilla-js-app` cookiecutter](../docker/templates/new-vanilla-js-app)
+is set up with a `develop` script that runs a local server and uses `watchify`
+to rebundle and retranspile JavaScript as it changes, as well as a `build`
+script to bundle and transpile code during deployment.
+
+#### Maintainability
+
+Gatsby continues to grow in popularity, and we don't anticipate that it will
+fall out of active support or development any time soon. With that said, its
+pre-requisite skills do make maintainability slightly more challenging as we
+cycle in new staff, who may or may not have those skills.
+
+Conversely, a lower learning curve means any staff could probably pick up a
+vanilla JavaScript app without issue.
+
+## When to use Vanilla ES6
+
+Does your site need:
+
+- Only one or two pages?
+- A minimally complicated map?
+- Lightweight data, e.g., a single JSON or CSV file or low cost external requests
+
+It could be well suited to the vanilla JavaScript stack! Read on for more
+details.
+
+On the other hand, does your site need:
+
+- Multiple pages, particularly pages based on data?
+- Heavyweight data, e.g., multiple data files, an external datastore (such as
+Hasura), or heavy external requests?
+- Complex maps, charts, or other visualizations?
+
+Then it's probably better suited for Gatsby. [Head over to our Gatsby docs to
+learn more](gatsby/).
+
+## Prerequisites
+
+For a maximally productive time, you'll want to familiarize yourself with
+[our JavaScript stack](https://github.com/datamade/how-to/blob/hcg/lil-js/javascript/stack.md),
+namely the `yarn` package manager and build tool, as well as ES6 itself.
+
+## Patterns
+
+### Development
+
+If your app doesn't have any dependencies, or if you can ship them with your
+code or source them from a CDN, simply create an `index.html` file and a
+`main.js` file, and off you go.
+
+You can serve your files in local development like:
+
+```bash
+python3 -m http.server
```

Comment:
more recently, I'm a fan of npm's `http-server`: https://www.npmjs.com/package/http-server

for example: `npx http-server . -c-1`